tests/models/t5/test_modeling_tf_t5.py

```diff
@@ -594,6 +573,43 @@ def test_sample_generate(self):
 
         self.assertListEqual(expected_output_string, output_strings)
 
+    @slow
+    def test_beam_search_xla_generate_simple(self):
+        model = TFT5ForConditionalGeneration.from_pretrained("t5-small")
+        tokenizer = T5Tokenizer.from_pretrained("t5-small")
+
+        # tests XLA with task specific arguments
+        task_specific_config = getattr(model.config, "task_specific_params", {})
+        translation_config = task_specific_config.get("translation_en_to_fr", {})
+        model.config.update(translation_config)
+
+        # two examples with different lengths to confirm that attention masks are operational in XLA
+        sentences = [
+            model.config.prefix + "Today is a beautiful day.",
+            model.config.prefix + "I have four cats, three dogs, two birds, and a horse.",
+        ]
+        input_ids = tokenizer(sentences, return_tensors="tf", padding=True).input_ids
+
+        # xla_generate = tf.function(model.generate, jit_compile=True)
+        xla_generate = tf.function(model.generate)
+
+        # TODO (joao): there is something not quite right with XLA T5 -- as we increase `max_length` the two outputs
+        # drift appart, where the XLA version clearly degrades its quality. XLA-related variables look fine (they are
```

Comment:
Let's also open an issue for this one after having merged the PR - my intuition would be that there is a problem with the relative position embeddings if you don't have the same problem in Bart. Relative position embeddings are quite tricky and could easily have been messed up by XLA